model/src/pyrenew/latent/infections.py

```diff
@@ -103,6 +103,10 @@ def sample(
         InfectionsSample
             Named tuple with "infections".
         """
+        if I0.size < gen_int.size:
+            raise ValueError(
+                f"Initial infections must be at least as long as the generation interval. Got {I0.size} initial infections and {gen_int.size} generation interval."
+            )
```

Comment:
This duplicates the lines right below (113-117), but I prefer doing the check here and removing the below.
Then  `recent_I0 = I0[-gen_int_rev.size :]` doesn't need to be wrapped in an `else`.

Also makes it more consistent with `InfectionsWithFeedback.sample()`. Which implies that for DRY purposes we should actually have a function that performes this bounds checking and subsetting. Separate PR though.

Reply:
addressed in https://github.com/CDCgov/multisignal-epi-inference/pull/124/commits/7a453a964466c0c27967a989e8fd7d673ad978ce